src/map/clif.c

```diff
@@ -18117,6 +18144,94 @@ void clif_party_leaderchanged(struct map_session_data *sd, int prev_leader_aid,
 	clif_send(buf, packet_len(0x7fc), &sd->bl, PARTY);
 }
 
+void clif_clan_message(struct clan *clan,const char *mes,int len){
+#if PACKETVER >= 20131223
+	struct map_session_data *sd;
+	uint8 buf[256];
+
+	if( len == 0 ){
+		return;
+	}else if( len > (sizeof(buf)-5-NAME_LENGTH) ){
+		ShowWarning("clif_clan_message: Truncated message '%s' (len=%d, max=%d, clan_id=%d).\n", mes, len, sizeof(buf)-5, clan->id);
+		len = sizeof(buf)-5-NAME_LENGTH;
+	}
+
+	WBUFW(buf, 0) = 0x98e;
+	WBUFW(buf, 2) = len + 5 + NAME_LENGTH;
+	
+	// Offially the sender name should also be filled here, but it is not required by the client and since it's in the message too we do not fill it
+	//safestrncpy((char*)WBUFP(buf,4), sendername, NAME_LENGTH);
+	safestrncpy((char*)WBUFP(buf,4+NAME_LENGTH), mes, len+1);
+
+	if((sd = clan_getavailablesd(clan)) != NULL)
+		clif_send(buf, WBUFW(buf,2), &sd->bl, CLAN);
+#endif
+}
+
+void clif_parse_clan_chat( int fd, struct map_session_data* sd ){
+#if PACKETVER >= 20131223
+	char name[NAME_LENGTH], message[CHAT_SIZE_MAX], output[CHAT_SIZE_MAX+NAME_LENGTH*2];
+
+	// validate packet and retrieve name and message
+	if( !clif_process_message(sd, false, name, message, output ) )
+		return;
+
+	clan_send_message( sd, (char*)RFIFOP(fd,4), RFIFOW(fd,2) - 4 );
+#endif
+}
+
+void clif_clan_basicinfo( struct map_session_data *sd ){
+#if PACKETVER >= 20131223
+	int fd = sd->fd, offset, length;
+	struct clan* clan = sd->clan;
+	char mapname[MAP_NAME_LENGTH_EXT];
+
```

Comment:
sure doesn't want to check the `sd` first?
and also the `fd` (`clif_session_isValid()` wow, forgot about this actually)
just incase if player is autorade, they has no `fd` set.